grpc/oob/machine/machine.go

```diff
@@ -225,24 +264,29 @@ func (m Action) PowerSet(ctx context.Context, action string) (result string, err
 	log.Info("connected to BMC", logMetadata(client.GetMetadata())...)
 	m.SendStatusMessage("connected to BMC")
 
+	// fetch the current power state that will be returned by "status"
+	// or used for control in cycle, and is always sent in traces
+	currentPowerState, err := client.GetPowerState(ctx)
+	if err != nil {
+		log.Error(err, "failed to get power state")
+		m.SendStatusMessage("error getting power state: " + err.Error())
+		return "", &repository.Error{
+			Code:    v1.Code_value["UNKNOWN"],
+			Message: err.Error(),
+		}
+	}
+
+	span.SetAttributes(attribute.String("currentPowerState", currentPowerState))
+
 	ok := true
 	if pwrAction == "status" {
-		result, err = client.GetPowerState(ctx)
+		result = currentPowerState
 	} else {
 		if action == v1.PowerAction_POWER_ACTION_CYCLE.String() {
 			// check status
 			// if powered on, do cycle
```

Comment:
It seems like this part of the comment may not have been accurate and remains inaccurate... Should it have been? Should we make it accurate (again?) ? CC @jacobweinstock